### PR TITLE
test login endpoint

### PR DESCRIPTION
**Pull request by matthewbub**
## Summary by CodeRabbit

* **New Features**
  * Login now supports authentication using either username or email address.
  * Enhanced test coverage for login functionality, including brute force and edge case scenarios.
  * Added a new SQL view to identify active users in the database.

* **Bug Fixes**
  * Improved email validation to reject inputs containing control characters.

* **Chores**
  * Refined cookie security settings for different environments.
  * Improved dynamic generation of test user accounts for testing.

* **Documentation**
  * Improved formatting and clarity in compliance documentation.

### CODE REVIEW

File: pkg/api/login.go [hunk 1]
```diff
@@ -93,17 +125,21 @@ func LoginHandler(c *gin.Context) {
...
 		}
-		if env == constants.ENV_STAGING || env == constants.ENV_DEVELOPMENT || env == constants.ENV_TEST {
+		if env == constants.ENV_STAGING {
+			cookieConfig.HttpOnly = true
+			cookieConfig.Secure = true
+		}
+		if env == constants.ENV_DEVELOPMENT || env == constants.ENV_TEST {
 			cookieConfig.HttpOnly = false
 			cookieConfig.Secure = false
 		}
```
[the remark this answers is not in the document]
Comment (matthewbub) [reply pkg/api/login.go:135]: this is to be expected for local developpment
